### PR TITLE
OGR2OGR: test JSON types roundtrip

### PR DESCRIPTION
**Pull request by elpaso**
This is a test I wrote when testing https://github.com/qgis/QGIS/issues/54966 and the issue fixed by @rouault with https://github.com/OSGeo/gdal/pull/8582

The code paths are probably already covered by unit tests in https://github.com/OSGeo/gdal/pull/8582/files#diff-9b2d54afcaf49a2c1a777b5a5a2025eef8dcf4b4fde3caaa17d649f1ac1d945dR2126 but maybe it can be added anyway.

### CODE REVIEW

File: autotest/utilities/test_ogr2ogr_lib.py [hunk 1]
```diff
@@ -2187,3 +2188,96 @@ def my_handler(errorClass, errno, msg):
...
+        assert fd.GetSubType() == ogr.OFSTNone
+
+    with gdal.ExceptionMgr(useExceptions=True):
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            src = os.path.join(temp_dir, "test_json.geojson")
+            dst = os.path.join(temp_dir, "test_json.gpkg")
+
+            with open(src, "w+") as f:
+                f.write(
```
Comment (rouault) [inline autotest/utilities/test_ogr2ogr_lib.py:2249]: should rather be something like (with indentation fixed):

```suggestion
            src = str(tmp_vsimem / "test_json.geojson")
            dst = str(tmp_vsimem / "test_json.gpkg")

           gdal.FileFromMemBuffer(src,
```

File: autotest/utilities/test_ogr2ogr_lib.py [hunk 2]
```diff
@@ -2187,3 +2188,96 @@ def my_handler(errorClass, errno, msg):
...
+
+            ds_output = gdal.VectorTranslate(
+                srcDS=src, destNameOrDestDS=dst, options=options
+            )
+            lyr = ds_output.GetLayerByName("test")
+
+            test_types(lyr)
+
+            # Write it back to json
+            round_trip_dst = os.path.join(temp_dir, "test_json_back.geojson")
```
Comment (rouault) [inline autotest/utilities/test_ogr2ogr_lib.py:2272]:
```suggestion
            round_trip_dst = str(tmp_vsimem / "test_json_back.geojson")
```

File: autotest/utilities/test_ogr2ogr_lib.py (hunk 1 above)
Comment (elpaso) [reply autotest/utilities/test_ogr2ogr_lib.py]: done

File: autotest/utilities/test_ogr2ogr_lib.py (hunk 2 above)
Comment (elpaso) [reply autotest/utilities/test_ogr2ogr_lib.py]: done

File: autotest/utilities/test_ogr2ogr_lib.py [hunk 3]
```diff
@@ -2190,6 +2190,99 @@ def my_handler(errorClass, errno, msg):
...
+            layerCreationOptions={"AUTODETECT_JSON_STRINGS": "FALSE"}
+        )
+        gdal.VectorTranslate(
+            srcDS=dst, destNameOrDestDS=round_trip_dst, options=options
+        )
+
+        with gdal.OpenEx(round_trip_dst, gdal.OF_VECTOR | gdal.OF_READONLY) as ds:
+            lyr = ds.GetLayer(0)
+            test_extended_types(lyr)
+
```
Comment (rouault) [inline autotest/utilities/test_ogr2ogr_lib.py:2282]:
```suggestion


```

File: autotest/utilities/test_ogr2ogr_lib.py [hunk 4]
```diff
@@ -2225,3 +2318,4 @@ def my_handler(errorClass, errno, msg):
                         )
                         is None
                     )
+
```
Comment (rouault) [inline autotest/utilities/test_ogr2ogr_lib.py:2321]:
```suggestion
```

**Pull request merged**